lib/tests/test_lexer.c

```diff
@@ -127,54 +109,84 @@ _current_lloc(void)
   return parser->yylloc;
 }
 
-#define assert_token_type(expected)                                     \
-  assert_gint(_current_token()->type, expected, "Bad token type at %s:%d", __FUNCTION__, __LINE__);
+static void
```

Comment:
The correct expansion of __FUNCTION__ and __LINE__ requires this assert to be a macro. With this change __FUNCTION__ would be replaced by `"assert_token_type"` unconditionally, losing the name of the calling function

Reply:
ok. Do I need to replace all assert functions as macros ?

Reply:
The guiding principle is to use functions whenever possible for sake of simple code. Though there are cases when macro is necessary: either because it is impossible to solve the problem with functions (see #1999 for example when the evaluation of the parameters must be delayed), or using macros would result in a simple/more understandable code. Your specific case belongs to the latter: one can resolve the expansion problem by passing `__FUNCTION__` and `__LINE__` as parameter to a regular function, but it would be inconvenient, and it increases the code complexity of the internal api.